pkg/controller/controller.go

```diff
@@ -339,14 +336,15 @@ func NewCSIProvisioner(client kubernetes.Interface,
 		identity:             identity,
 		volumeNamePrefix:     volumeNamePrefix,
 		volumeNameUUIDLength: volumeNameUUIDLength,
+		driverName:           driverName,
 	}
 	return provisioner
 }
 
 // This function get called before any attempt to communicate with the driver.
 // Before initiating Create/Delete API calls provisioner checks if Capabilities:
 // PluginControllerService,  ControllerCreateVolume sre supported and gets the  driver name.
-func checkDriverState(grpcClient *grpc.ClientConn, timeout time.Duration, needSnapshotSupport bool) (*driverState, error) {
+func checkDriverCapabilities(grpcClient *grpc.ClientConn, timeout time.Duration, needSnapshotSupport bool) (sets.Int, error) {
```

Comment:
Why are we representing capabilities with `sets.Int`, is there no named type from the protobuf we can use? if not, can we define one
`type PluginCapabilitiesType sets.Int`

Reply:
I tried, but then I loose all `sets.Int` methods (no `PluginCapabilitiesType.Has`). I could restore `driverState` struct (with a better name) with `sets.Int` as a field, but that looks ugly.